## STUDY CASE (REVISION): Review of Various Products Manufactured by Amazon

Objective
1. To use these data, explore it, and generate insights from it.


### Basic Text Data Pre-processing

In [1]:
import numpy as np
import pandas as pd
import os
# for visualizations
import matplotlib.pyplot as plt
# for regular expressions
import re
# for handling string
import string
# for performing mathematical operations
import math

In [2]:
 # Importing dataset
df = pd.read_csv("dataset.csv")

C:\Users\User\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (1,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
pwd

'C:\\Users\\User\\Desktop\\Resume\\Projects\\Amazon'

In [4]:
df = pd.read_csv("dataset.csv")
print("Shape of data=>",df.shape)

Shape of data=> (34660, 21)


The dataset contains 34660 rows and 21 columns. We will use the only information that we need so we reducing the dataset to only four columns: ‘name’, ‘reviews.text’, ‘reviews.doRecommend’, and ‘reviews.numHelpful’.

In [6]:
df = df[['name','reviews.text','reviews.doRecommend','reviews.numHelpful']]
print("Shape of data=>",df.shape)
df.head(5)

Shape of data=> (34660, 4)


name  \
0  All-New Fire HD 8 Tablet, 8 HD Display, Wi-Fi,...   
1  All-New Fire HD 8 Tablet, 8 HD Display, Wi-Fi,...   
2  All-New Fire HD 8 Tablet, 8 HD Display, Wi-Fi,...   
3  All-New Fire HD 8 Tablet, 8 HD Display, Wi-Fi,...   
4  All-New Fire HD 8 Tablet, 8 HD Display, Wi-Fi,...   

                                        reviews.text reviews.doRecommend  \
0  This product so far has not disappointed. My c...                True   
1  great for beginner or experienced person. Boug...                True   
2  Inexpensive tablet for him to use and learn on...                True   
3  I've had my Fire HD 8 two weeks now and I love...                True   
4  I bought this for my grand daughter when she c...                True   

   reviews.numHelpful  
0                 0.0  
1                 0.0  
2                 0.0  
3                 0.0  
4                 0.0

In [7]:
# Let's check if there are any null values.
df.isnull().sum()

name                   6760
reviews.text              1
reviews.doRecommend     594
reviews.numHelpful      529
dtype: int64

There are a few null values in the dataset. So let's drop the null values.

In [8]:
df.dropna(inplace=True)
df.isnull().sum()

name                   0
reviews.text           0
reviews.doRecommend    0
reviews.numHelpful     0
dtype: int64

We will only considering those products that have at least 500 reviews. We doing this to make sure that we have a sufficient number of reviews for each product. Here, we will use lambda functions with filter() to filter our dataset.

In [9]:
df = df.groupby('name').filter(lambda x:len(x)>500).reset_index(drop=True)
print('Number of products=>',len(df['name'].unique()))

Number of products=> 8


Now, we are left with eight products. Also, the ‘reviews.doRecommend’ column contains values in the form of True-False and’ reviews.numHelpful’ contains floating-point numbers, which is not possible. Therefore, we need to convert these columns into integers:

In [10]:
df['reviews.doRecommend']=df['reviews.doRecommend'].astype(int)
df['reviews.numHelpful']=df['reviews.numHelpful'].astype(int)